onnxruntime/core/session/inference_session.cc

```diff
@@ -422,6 +422,24 @@ common::Status InferenceSession::Initialize() {
                                                    std::make_unique<CPUExecutionProvider>(epi)));
     }
 
+    if (!session_options_.enable_sequential_execution) {
+      for (auto it = execution_providers_.begin(); it != execution_providers_.end(); ++it) {
+        // very rare case - user has registered CPU provider first -
+        // so parallel execution is still possible as all nodes will get assigned CPU provider
+        if (it->get()->Type() == onnxruntime::kCpuExecutionProvider) {
+          break;
+        }
+
+        if (it->get()->Type() == onnxruntime::kCudaExecutionProvider) {
+          LOGS(*session_logger_, ERROR) << "Parallel execution cannot be enabled "
+                                           "when CUDA execution provider is registered.";
```

Comment:
This is not accurate, right? If CPU is registered first and CUDA as second, this error won't be thrown.

Reply:
That was the intention. If CPU is registered first, won't all the nodes be assigned to CPU anyway? If all nodes are assigned to CPU, parallel execution is possible. In that case it doesn't matter if CUDA provider is registered (after the CPU provider).

Reply:
I meant the error message is slightly inaccurate. In both cases cuda is registered, but we're enabling parallel executor in one case, but not in another. May be it should read - "Parallel execution cannot be enabled when CUDA execution provider is registered *as first priority*."

Reply:
Will modify